### ISSUE TITLE
Support of the `@OneOf` Input Object feature

### ISSUE THREAD

**Issue by lucarin91** [graphql-rust/juniper#1062]
Is there any plan for implementing the [RFC: OneOf Input Objects](https://github.com/graphql/graphql-spec/pull/825)? It is not a final draft but it is getting a lot of attention and other GraphQL libraries already started to implement that.

I am also willing to contribute to the feature if you think it will be useful.

**Comment by tyranron** [comment]
@lucarin91 need to think about this. I'm quite unsure about the RFC itself. It looks like copy-paster attempt from Protobuf, but "why the hell" when we have unions in GraphQL? Seems like input unions consisting of input objects (optionally, scalars too) is the thing fitting more naturally into GraphQL.

**Comment by lucarin91** [comment]
> @lucarin91 need to think about this. I'm quite unsure about the RFC itself. It looks like copy-paster attempt from Protobuf, but "why the hell" when we have unions in GraphQL? Seems like input unions consisting of input objects (optionally, scalars too) is the thing fitting more naturally into GraphQL.

Yeah, I agree it isn't the cleaner solution, but for my limited understanding, they came up with that after the better solution, i.e., the *tagged* types, was not accepted. The [*tagged* types](https://github.com/graphql/graphql-spec/pull/733) were a general way to have unions on inputs and outputs, but I think there was concern on the implementation side because it will greatly change the language. 

I am interested in this feature in order to have an input object that will be parsed as a rust enum with struct variants. I managed to implement that as a plain GraphQL object but I will not be able to have validation. So the user could provide every variant at once.

Do you think it will be a complex feature to add the support of this `@OneOf` directive?

**Comment by ilslv** [comment]
@tyranron isn't this request basically the same as https://github.com/graphql-rust/juniper/issues/1055 but with added directive, that should be generated in the output (introspection or GraphQL API definition)?

**Comment by devtempmac** [comment]
quoting https://github.com/graphql/graphql-spec/pull/825#issuecomment-2660789344: 
> In GraphQL js 15/16 you can use it unflagged and same goes for GraphQL ruby so I'd say it's stable for implementations. It's also in stage 2 which seems a bit more solidified in terms of how it will work

@tyranron is `oneof` still a no-no in terms of stability?
(2 years have passed since the last discussion on this issue, 
and I'm wondering if your opinion on this issue has changed)
